components/constellation/constellation.rs

```diff
@@ -2043,42 +2043,37 @@ where
             return warn!("Browsing context group not found");
         };
         let webgpu_chan = match browsing_context_group.webgpus.entry(host) {
-            Entry::Vacant(v) => v
-                .insert(
-                    match WebGPU::new(
-                        self.webrender_wgpu.webrender_api.create_sender(),
-                        self.webrender_document,
-                        self.webrender_wgpu.webrender_external_images.clone(),
-                        self.webrender_wgpu.wgpu_image_map.clone(),
-                    ) {
-                        Some(webgpu) => {
-                            let msg = ConstellationControlMsg::SetWebGPUPort(webgpu.1);
-                            if let Err(e) = source_pipeline.event_loop.send(msg) {
-                                warn!(
-                                    "{}: Failed to send SetWebGPUPort to pipeline ({:?})",
-                                    source_pipeline_id, e
-                                );
-                            }
-                            webgpu.0
-                        },
-                        None => {
-                            return warn!("Failed to create new WebGPU thread");
-                        },
-                    },
-                )
-                .clone(),
-            Entry::Occupied(o) => o.get().clone(),
+            Entry::Vacant(v) => WebGPU::new(
+                self.webrender_wgpu.webrender_api.create_sender(),
+                self.webrender_document,
+                self.webrender_wgpu.webrender_external_images.clone(),
+                self.webrender_wgpu.wgpu_image_map.clone(),
+            )
+            .map(|webgpu| {
+                let msg = ConstellationControlMsg::SetWebGPUPort(webgpu.1);
+                if let Err(e) = source_pipeline.event_loop.send(msg) {
+                    warn!(
+                        "{}: Failed to send SetWebGPUPort to pipeline ({:?})",
+                        source_pipeline_id, e
+                    );
+                }
+                v.insert(webgpu.0).clone()
+            }),
+            Entry::Occupied(o) => Some(o.get().clone()),
         };
         match request {
-            FromScriptMsg::RequestAdapter(response_sender, options, ids) => {
-                let adapter_request = WebGPURequest::RequestAdapter {
-                    sender: response_sender,
-                    options,
-                    ids,
-                };
-                if webgpu_chan.0.send((None, adapter_request)).is_err() {
-                    return warn!("Failed to send request adapter message on WebGPU channel");
-                }
+            FromScriptMsg::RequestAdapter(response_sender, options, ids) => match webgpu_chan {
```

Comment:
So in this PR you want to update not the handling of this message, but of the `FromScriptMsg::GetWebGPUChan` below. 

Updating this message too would be a nice follow-up actually. You could open an issue, and then a PR for it too.

Reply:
So, I'm not allowed to change the handling of `FromScriptMsg::RequestAdapter` ??  I don't think I can do that and get it to compile.

Reply:
Ah yes, I see what you mean, and actually the other message handling doesn't need to change because the option can be passed through as is... thanks for pointing it out!